modules/carousel/jetpack-carousel.js

```diff
@@ -993,7 +993,7 @@ jQuery(document).ready(function($) {
 			var imageLinkParser = document.createElement( 'a' );
 			imageLinkParser.href = args.large_file;
 
-			var isPhotonUrl = ( imageLinkParser.hostname.match( /^i[\d]{1}.wp.com$/i ) != null );
+			var isPhotonUrl = /^i[\d]{1}.wp.com$/i.test( imageLinkParser.hostname );
```

Comment:
This could be more strict, photon urls should only match `i[0-2]`. Also `[\d]{1}` should be the same as `[\d]` which should be the same as `\d`.

```suggestion
			var isPhotonUrl = /^i[0-2].wp.com$/i.test( imageLinkParser.hostname );
```

```js
function before( input ) {
  return input.match( /^i[\d]{1}.wp.com$/i ) != null;
}
function after( input ) {
  return /^i[0-2].wp.com$/i.test( input );
}

const candidates = [
   's0.wP.com',
   'I2.wp.COM',
   'example.wp.com',
   'jetpack.com',
   'I0.wp.COM',
   'I1.WP.COM',
   'i2.wp.com',
]

candidates.every( test => before( test ) === after( test ) );
```

Reply:
Make sense :+1:

Reply:
Prefer the updated regex as well?